doc/test_messages_documentation.py

```diff
@@ -151,13 +149,29 @@ def _runTest(self) -> None:
         expected_messages = self._get_expected()
         actual_messages = self._get_actual()
         if self.is_good_test_file():
-            msg = "There should be no warning raised for 'good.py'"
-            assert actual_messages.total() == 0, msg  # type: ignore[attr-defined]
+            assert actual_messages.total() == 0, self.assert_message_good(actual_messages)  # type: ignore[attr-defined]
         if self.is_bad_test_file():
             msg = "There should be at least one warning raised for 'bad.py'"
             assert actual_messages.total() > 0, msg  # type: ignore[attr-defined]
         assert expected_messages == actual_messages
 
+    def assert_message_good(self, actual_messages: MessageCounter) -> str:
+        messages = "\n- ".join(f"{v} (l. {i})" for i, v in actual_messages)
+        msg = f"""There should be no warning raised for 'good.py' but these messages were raised:
+- {messages}
+
+See:
+
+"""
+        lines = []
+        with open(self._test_file[1]) as f:
+            lines = f.readlines()
+        for actual_message in actual_messages:
+            line, value = actual_message
+            new_value = f"{lines[line-1][:-1]}  # <-- /!\\ unexpected '{value}' /!\\\n"
+            lines[line - 1] = new_value
```

Comment:
```suggestion
            lines[line - 1] = f"{lines[line-1][:-1]}  # <-- /!\\ unexpected '{value}' /!\\\n"
```

Reply:
I didn't do that but refactored the ``with open`` a little, I'll let you check I think it's better.